test/api/small-catches.unit.test.js

```diff
@@ -0,0 +1,251 @@
+const SmallCatchesApi = require('../../src/api/small-catches')
+const ActivityApi = require('../../src/api/activities')
+const RiversApi = require('../../src/api/rivers')
+const MethodsApi = require('../../src/api/methods')
+const EntityApi = require('../../src/api/entity-api')
+const { monthHelper } = require('../../src/handlers/common')
+const ResponseError = require('../../src/handlers/response-error')
+
+jest.mock('../../src/api/activities')
+jest.mock('../../src/api/rivers')
+jest.mock('../../src/api/methods')
+jest.mock('../../src/handlers/common', () => ({
+  monthHelper: {
+    find: {
+      numFromKey: jest.fn(),
+      keyFromNum: jest.fn()
+    }
+  }
+}))
+jest.mock('../../src/handlers/response-error', () => ({
+  status: {
+    CONFLICT: 409
+  }
+}))
+
+describe('small-catches.unit', () => {
+  describe('constructor', () => {
+    it('maps API response into formatted object', async () => {
+      const fakeActivity = {
+        id: '1',
+        days: 3,
+        _links: {
+          river: {
+            href: 'https://local/api/rivers/2'
+          },
+          self: {
+            href: 'https://local/api/activities/1'
+          }
+        }
+      }
+      const fakeRiver = {
+        id: '2',
+        name: 'Avon',
+        _links: {
+          self: {
+            href: 'https://local/api/rivers/2'
+          }
+        }
+      }
+      const fakeMethods = [
+        { id: 'M1', name: 'Fly', internal: false },
+        { id: 'M2', name: 'Spin', internal: true }
+      ]
+
+      ActivityApi.prototype.getFromLink.mockResolvedValue(fakeActivity)
+      RiversApi.prototype.getFromLink.mockResolvedValue(fakeRiver)
+      MethodsApi.prototype.list.mockResolvedValue(fakeMethods)
+      MethodsApi.prototype.getFromLink.mockImplementation(async (_req, href) => ({ id: href }))
+      MethodsApi.prototype.doMap.mockResolvedValue({ name: 'Fly' })
+
+      monthHelper.find.numFromKey.mockReturnValue(6)
```

Comment:
Would prefer to `mockResolvedValueOnce` if we can (or a set number of times) to avoid spillover to other tests.